index.bs

```diff
@@ -10518,126 +10522,140 @@ The characteristics of this property are as follows:
-        1.  Let |V| be the value of the first argument passed to the <emu-val>Function</emu-val>.
-        1.  If the [=attribute=] is a [=regular attribute=], then:
-            1.  Let |I| be the [=interface=]
-                whose [=interface prototype object=]
-                this property corresponding to the attribute appears on.
-            1.  Let |O| be the <emu-val>this</emu-val> value.
-            1.  If |O| is a [=platform object=],
-                then [=perform a security check=], passing:
-                *   the platform object |O|,
-                *   the [=identifier=] of the [=attribute=], and
-                *   the type “setter”.
-            1.  Let |validThis| be <emu-val>true</emu-val> if |O| is a
-                [=platform object=] that implements |I|, or
-                <emu-val>false</emu-val> otherwise.
-            1.  If |validThis| is <emu-val>false</emu-val> and the
-                [=attribute=] was not specified with the
-                [{{LenientThis}}] [=extended attribute=],
-                then [=ECMAScript/throw=] a <emu-val>TypeError</emu-val>.
-            1.  If the attribute is declared with a [{{Replaceable}}]
-                extended attribute, then:
-                1.  Let |P| be the identifier of the attribute.
-                1.  Call [=CreateDataProperty=](|O|, |P|, |V|).
+        1.  If |attribute|'s type is a [=promise type=], then:
+            1.  Let |reject| be the initial value of [=%Promise%=].reject.
+            1.  Return the result of calling |reject| with [=%Promise%=] as the
+                <emu-val>this</emu-val> object and the exception as the single argument value.
+        1.  Otherwise, end these steps and allow the exception to propagate.
+    1.  Let |F| be [=!=] [=CreateBuiltinFunction=](|realm|,
+        |steps|, the [=%FunctionPrototype%=] of |realm|).
+    1.  Let |name| be the string "get " prepended to |attribute|'s [=identifier=].
+    1.  Perform [=!=] [=SetFunctionName=](|F|, |name|).
+    1.  Perform [=!=] [=DefinePropertyOrThrow=](|F|, "length",
+        PropertyDescriptor<span class="prop-desc">{\[[Value]]: 0, \[[Writable]]: <emu-val>false</emu-val>, \[[Enumerable]]: <emu-val>false</emu-val>, \[[Configurable]]: <emu-val>true</emu-val>}</span>).
+    1. Return |F|.
+
+</div>
+
+<div algorithm>
+
+    The <dfn id="dfn-attribute-setter" export>attribute setter</dfn> is created as follows, given an
+    [=attribute=] |attribute|, an [=interface=] |target|, and a [=Realm=] |realm|:
+
+    1.  If |attribute| is [=read only=] and does not have a
+        [{{LenientThis}}], [{{PutForwards}}] or [{{Replaceable}}] [=extended attribute=], return
+        <emu-val>undefined</emu-val>; there is no [=attribute setter=] function.
+    1.  Assert: |attribute|'s type is not a [=promise type=].
+    1.  Let |steps| be the following series of steps:
+        1.  If no arguments were passed, then
+        1.  Let |V| be the value of the first argument passed.
```

Comment:
There's something unclear in the above two steps (either there's no argument or there is one).